core/src/main/java/tech/tablesaw/io/csv/CsvReader.java

```diff
@@ -207,64 +207,69 @@ public static Table read(InputStream stream, String tableName, boolean header, c
-                             String tableName,
-                             ColumnType[] types,
-                             boolean header,
-                             char columnSeparator) throws IOException {
-      BufferedReader streamReader = new BufferedReader(new InputStreamReader(stream));
-
-      Table table;
-      CSVParser csvParser = new CSVParserBuilder()
-          .withSeparator(columnSeparator)
-          .build();
-      try (CSVReader reader = new CSVReaderBuilder(streamReader).withCSVParser(csvParser).build()) {
-
-          String[] nextLine;
-          String[] columnNames;
-          List<String> headerRow;
-          if (header) {
-              nextLine = reader.readNext();
-              headerRow = Lists.newArrayList(nextLine);
-              columnNames = selectColumnNames(headerRow, types);
-          } else {
-              columnNames = makeColumnNames(types);
-              headerRow = Lists.newArrayList(columnNames);
-          }
-
-          table = Table.create(tableName);
-          for (int x = 0; x < types.length; x++) {
-              if (types[x] != ColumnType.SKIP) {
-                  String columnName = headerRow.get(x);
-                  if (Strings.isNullOrEmpty(columnName)) {
-                      columnName = "Column " + table.columnCount();
-                  }
-                  Column newColumn = TypeUtils.newColumn(columnName.trim(), types[x]);
-                  table.addColumn(newColumn);
-              }
-          }
-          int[] columnIndexes = new int[columnNames.length];
-          for (int i = 0; i < columnIndexes.length; i++) {
-              // get the index in the original table, which includes skipped fields
-              columnIndexes[i] = headerRow.indexOf(columnNames[i]);
-          }
-          // Add the rows
-          long rowNumber = header ? 1L : 0L;
-          while ((nextLine = reader.readNext()) != null) {
-              // for each column that we're including (not skipping)
-              int cellIndex = 0;
-              for (int columnIndex : columnIndexes) {
-                  Column column = table.column(cellIndex);
-                  try {
-                      column.appendCell(nextLine[columnIndex]);
-                  } catch (Exception e) {
-                      throw new AddCellToColumnException(e, columnIndex, rowNumber, columnNames, nextLine);
-                  }
-                  cellIndex++;
-              }
-              rowNumber++;
-          }
-      }
-      return table;
+            String tableName,
+            ColumnType[] types,
+            boolean header,
+            char columnSeparator) throws IOException {
+
+        // All other read methods end up here, make sure we do not have leading Unicode BOM
+        UnicodeBOMInputStream ubis = new UnicodeBOMInputStream(stream);
+        ubis.skipBOM();
+
+        BufferedReader streamReader = new BufferedReader(new InputStreamReader(ubis));
```

Comment:
we shouldn't need to wrap the `InputStreamReader` in a `BufferedReader`. I would just inline this below:

    try (CSVReader reader = new CSVReaderBuilder(new InputStreamReader(ubis)).withCSVParser(csvParser).build()) {